tests/unit/test_sql_sensor.py

```diff
@@ -6,149 +6,104 @@
 
 
 class ClickHouseSqlSensorTestCase(unittest.TestCase):
-    @mock.patch(
-        'airflow_clickhouse_plugin.sensors.clickhouse_sql_sensor.ClickHouseHook'
-    )
-    def test_clickhouse_sql_sensor_poke(self, mock_hook):
-        op = ClickHouseSqlSensor(task_id='sql_sensor_check', sql='SELECT 1',)
-
-        mock_get_records = mock_hook().get_records
-
-        mock_get_records.return_value = []
-        self.assertFalse(op.poke(None))
-
-        mock_get_records.return_value = [[None]]
-        self.assertTrue(op.poke(None))
-
-        mock_get_records.return_value = [['None']]
-        self.assertTrue(op.poke(None))
-
-        mock_get_records.return_value = [[0.0]]
-        self.assertTrue(op.poke(None))
-
-        mock_get_records.return_value = [[0]]
-        self.assertFalse(op.poke(None))
-
-        mock_get_records.return_value = [['0']]
-        self.assertFalse(op.poke(None))
-
-        mock_get_records.return_value = [['1']]
-        self.assertTrue(op.poke(None))
-
-    @mock.patch(
-        'airflow_clickhouse_plugin.sensors.clickhouse_sql_sensor.ClickHouseHook'
-    )
-    def test_clickhouse_sql_sensor_poke_fail_on_empty(self, mock_hook):
-        op = ClickHouseSqlSensor(
-            task_id='sql_sensor_check', sql='SELECT 1', fail_on_empty=True,
-        )
-
-        mock_get_records = mock_hook().get_records
-
-        mock_get_records.return_value = []
-        self.assertRaises(AirflowException, op.poke, None)
-
-    @mock.patch(
-        'airflow_clickhouse_plugin.sensors.clickhouse_sql_sensor.ClickHouseHook'
-    )
-    def test_clickhouse_sql_sensor_poke_success(self, mock_hook):
-        op = ClickHouseSqlSensor(
-            task_id='sql_sensor_check', sql='SELECT 1', success=lambda x: x in [1],
-        )
-
-        mock_get_records = mock_hook().get_records
-
-        mock_get_records.return_value = []
-        self.assertFalse(op.poke(None))
-
-        mock_get_records.return_value = [[1]]
-        self.assertTrue(op.poke(None))
-
-        mock_get_records.return_value = [['1']]
-        self.assertFalse(op.poke(None))
-
-    @mock.patch(
-        'airflow_clickhouse_plugin.sensors.clickhouse_sql_sensor.ClickHouseHook'
-    )
-    def test_clickhouse_sql_sensor_poke_failure(self, mock_hook):
+    def test_clickhouse_sql_sensor_poke(self):
+        op = ClickHouseSqlSensor(task_id='_', sql='',)
+        for return_value, expected_result in (
+                ([], False),
+                ([[None]], True),
+                ([['None']], True),
+                ([[0.0]], True),
+                ([[0]], False),
+                ([['0']], False),
+                ([['1']], True),
+        ):
```

Comment:
Just want to note that ```[], [[0.0]], [["0"]]```
gives the opposite result on Airflow 1.10.10 and possibly >=1.10.7

Reply:
Yeah, the logic has changed:
* 1.10.6, just `str`: https://github.com/apache/airflow/blob/1.10.6/airflow/sensors/sql_sensor.py#L102
* 1.10.7, `bool`: https://github.com/apache/airflow/blob/1.10.7/airflow/sensors/sql_sensor.py#L111